core/driver/sources/src/main/scala/org/virtuslab/ideprobe/config/IdeaProperties.scala

```diff
@@ -0,0 +1,28 @@
+package org.virtuslab.ideprobe.config
+
+import pureconfig.ConfigReader
+import pureconfig.generic.auto._
+import pureconfig.generic.ProductHint
+
+import java.nio.file.Path
+
+object IdeaProperties {
```

Comment:
If this is part of configuring ide-probe, I don't think we need to mimic IJ model. If this is just for sake of parsing idea.properties, I think it is a bit of overkill.
Just `Config.fromFile(...).get[String]("java.util.prefs.userRoot")` would do the job.